src/LooksRare.ts

```diff
@@ -234,6 +236,48 @@ export class LooksRare {
     };
   }
 
+  /**
+   * Create a maker bid for collection offer.
+   * @see this.createMakerBid
+   * @param orderInputs Order data
+   * @returns CreateMakerBidOutput
+   */
+  public createMakerCollectionOffer(orderInputs: CreateMakerCollectionOfferInput): Promise<CreateMakerBidOutput> {
+    return this.createMakerBid({ ...orderInputs, strategyId: StrategyType.collection, itemIds: [] });
+  }
+
+  public async createMakerCollectionOfferWithMerkleTree(
+    orderInputs: CreateMakerCollectionOfferInput,
+    itemIds: BigNumberish[]
+  ): Promise<CreateMakerBidOutput & { proofs: { itemId: BigNumberish; proof: string[] }[] }> {
+    const leaves = itemIds.map((itemId) => {
+      const hash = utils.keccak256(utils.solidityPack(["uint256"], [itemId]));
+      return Buffer.from(hash.slice(2), "hex");
+    });
+    const tree = new MerkleTreeJS(leaves, keccak256, { sortPairs: true });
+    const root = tree.getHexRoot();
+
+    const output = await this.createMakerBid({
+      ...orderInputs,
+      strategyId: StrategyType.collectionWithMerkleTree,
+      additionalParameters: [root],
+      itemIds: [],
+    });
+
+    const proofs = itemIds.map((itemId, index) => {
```

Comment:
proofs are only required during taker execution so I wonder if it is necessary at this point?

Reply:
If I don't compute it here, it mean we need to pass the full list of ids to the create taker function, not sure what's best

Reply:
Updated, I think it's better 👍